identity/src/main/java/com/stripe/android/identity/ui/BottomsheetHTML.kt

```diff
@@ -0,0 +1,65 @@
+package com.stripe.android.identity.ui
+
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDecoration
+import com.stripe.android.identity.networking.STRIPE_BOTTOM_SHEET
+import com.stripe.android.identity.networking.models.VerificationPageStaticContentBottomSheetContent
+import com.stripe.android.identity.viewmodel.BottomSheetViewModel
+import com.stripe.android.uicore.text.HtmlWithCustomOnClick
+
+/**
+ * Draw Html with the ability to open a web link or bottomsheet
+ */
+@Composable
+@ExperimentalMaterialApi
+internal fun BottomSheetHTML(
+    html: String,
+    modifier: Modifier = Modifier,
+    bottomSheetViewModel: BottomSheetViewModel,
+    bottomSheets: Map<String, VerificationPageStaticContentBottomSheetContent>?,
+    color: Color = Color.Unspecified,
+    style: TextStyle,
+    urlSpanStyle: SpanStyle = SpanStyle(textDecoration = TextDecoration.Underline)
+) {
+    val context = LocalContext.current
+    HtmlWithCustomOnClick(
+        html = html,
+        modifier = modifier,
+        color = color,
+        style = style,
+        urlSpanStyle = urlSpanStyle
+    ) { annotatedStringRanges ->
+        annotatedStringRanges.firstOrNull()?.let {
```

Comment:
on Connections we have this same logic. We handle this in the Viewmodel to be able to test it. 

https://github.com/stripe/stripe-android/blob/854e81c5acf6c4ffde0268054ae57083ab519608/financial-connections/src/main/java/com/stripe/android/financialconnections/features/consent/ConsentViewModel.kt#L82

Reply:
interesting, it seems this viewmodel has dedicated LaunchedEffect inside ConsentScreen when url is parsed. For Identity the url is not a set of values and not only applicable to one screen(please refer to [this](https://docs.google.com/document/d/1m74HPjVrhHnuRdl8sda1wjRpj33ULyFoupk4xPibx7s/edit#bookmark=id.rlmy5dk2zz2o) section on how we are defining the bottomsheets), so I thin we would ty to make `BottomSheetHTML` context agnostic